website/static/js/oauthAddonNodeConfig.js

```diff
@@ -0,0 +1,244 @@
+/**
+ * Module that controls the addon node settings. Includes Knockout view-model
+ * for syncing data, and HGrid-folderpicker for selecting a folder.
+ */
+'use strict';
+
+var ko = require('knockout');
+require('knockout.punches');
+var $ = require('jquery');
+var Raven = require('raven-js');
+var bootbox = require('bootbox');
+
+
+var ZeroClipboard = require('zeroclipboard');
+ZeroClipboard.config('/static/vendor/bower_components/zeroclipboard/dist/ZeroClipboard.swf');
+var $osf = require('js/osfHelpers');
+var oop = require('js/oop');
+var FolderPickerViewModel = require('js/folderPickerNodeConfig');
+
+ko.punches.enableAll();
+
+/**
+ * View model to support instances of AddonNodeConfig (folder picker widget)
+ *
+ * @class AddonFolderPickerViewModel
+ * @param {String} addonName Full display name of the addon
+ * @param {String} url API url to initially fetch settings
+ * @param {String} selector CSS selector for containing div
+ * @param {String} folderPicker CSS selector for folderPicker div
+ * @param {Object} opts Optional overrides to the class' default treebeardOptions, in particular onPickFolder
+ */
+var OauthAddonFolderPickerViewModel = oop.extend(FolderPickerViewModel, {
+    constructor: function(addonName, url, selector, folderPicker, opts) {
+        var self = this;
+        self.super.constructor.call(self, addonName, url, selector, folderPicker);
+        // whether the auth token is valid
+        self.validCredentials = ko.observable(true);
+        // Emails of contributors, can only be populated by activating the share dialog
+        self.emails = ko.observableArray([]);
+        self.loading = ko.observable(false);
+        // Whether the contributor emails have been loaded from the server
+        self.loadedEmails = ko.observable(false);
+        // externalAccounts
+        self.accounts = ko.observable([]);
+        // List of contributor emails as a comma-separated values
+        self.emailList = ko.pureComputed(function() {
+            return self.emails().join([', ']);
+        });
+        self.selectedFolderType = ko.pureComputed(function() {
+            var userHasAuth = self.userHasAuth();
+            var selected = self.selected();
+            return (userHasAuth && selected) ? selected.type : '';
+        });
+        self.messages.submitSettingsSuccess =  ko.pureComputed(function() {
+            return 'Successfully linked "' + $osf.htmlEscape(self.folder().name) + '". Go to the <a href="' +
```

Comment:
Building HTML in knockout ViewModels is strongly discouraged. All HTML should be handled in the template, and knockout should be allowed to handle escaping of custom values there.

http://knockoutjs.com/documentation/observables.html#mvvm-and-view-models
```
A view model: a pure-code representation of the data and operations on a UI. ...  When using KO, your view models are pure JavaScript objects that hold no knowledge of HTML. 
```